OutOfSchool/OutOfSchool.WebApi/Common/Responses/ApiErrorResponse.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Ardalis.SmartEnum;
+
+namespace OutOfSchool.WebApi.Common.Responses;
+
+public record ApiErrorResponse
+{
+    private ApiErrorResponse(string groupName, string code, string message)
+    {
+        GroupName = groupName ?? throw new ArgumentNullException(nameof(groupName));
+        Code = code ?? throw new ArgumentNullException(nameof(code));
+        Message = message;
+    }
+
+    public static ApiErrorResponse Create(string groupName, string code, string message)
```

Comment:
Please explain why instead of public ctor you have private ctor and public static? Statics are worser from testability point of view. As it does the same as ctor - what is the main reason to have it?